### PR TITLE
tests/riotboot: migrate to full python test script

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR adapts `tests/riotboot` automatic test to a single python script. It also eliminate the need to fix an APP_VER=0.

### Testing procedure

- Run:

`make -C tests/riotboot flash test`

- Verify CI output.

### Issues/PRs references

Expands on #11707.

### CODE REVIEW

**Comment by fjmolinas** [comment]
@kaspar030 I have an Issue where running tests doesn't seem deterministic. I re-launched the tests although last run was successful, and although no code changes have been made, some tests failed this time, interestingly not on the same platform and in one case it succeeded on gnu and the other on lvm. Could it be that the term output isn't getting captured properly? It seems it detects the wrong image version at times.
 
Also I have seen `tests/nordic_softdevice` fail a couple of times, seen also in #11956.

**Comment by fjmolinas** [comment]
On a side note could it be possible to add a test progress bar to murdock? its confusing to see the completion at 100% at have it go on for 20 more minutes of testing.

**Comment by miri64** [comment]
> On a side note could it be possible to add a test progress bar to murdock? its confusing to see the completion at 100% at have it go on for 20 more minutes of testing.

I think it's more visible if you write these kind of requests to ci@riot-os.org ;-)

**Comment by fjmolinas** [comment]
> I think it's more visible if you write these kind of requests to ci@riot-os.org ;-)

hmmm.. for some reason the email is bouncing back.

File: tests/riotboot/tests/01-run.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    child.sendline("curslothdr")
+    child.expect(r"Image Version: (?P<app_ver>0x[0-9a-fA-F]{8})\r\n")
+    current_app_ver = int(child.match.group("app_ver"), 16)
+    child.expect_exact('>')
+
+    # Flash to both slots and verify basic functions
+    for version in [current_app_ver + 1, current_app_ver + 2]:
+        slotnum = slotnum ^ 1
+        flash_slot(slotnum, version)
+        assert_slot_num(child, slotnum, version)
```
Comment (aabadie) [inline tests/riotboot/tests/01-run.py]: This function is not only checking the slot num, like its name would suggest but also the application version, the slot addr, etc. Maybe rename it to `_check_slot` or something ?

**Comment by fjmolinas** [comment]
> A few minor comments, otherwise that looks good!

Can I squash directly?

**Comment by aabadie** [comment]
> Can I squash directly?

Sure!

File: tests/riotboot/tests/01-run.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    child.expect_exact('>')
+    child.sendline("curslotnr")
+    child.expect_exact("Current slot={}".format(slotnum))
+
+    # Ask for current slot header info and checks for basic output integrity
+    child.expect_exact('>')
+    child.sendline("curslothdr")
+    # Magic number is "RIOT" (always in little endian)
+    child.expect_exact("Image magic_number: 0x544f4952")
+    # Other info is hardware/app dependant so we just check basic compliance
```
Comment (aabadie) [inline tests/riotboot/tests/01-run.py]:
```suggestion
    # Other info is hardware/app dependent so we just check basic compliance
```

Comment (aabadie) [reply tests/riotboot/tests/01-run.py]: Thanks codespell :)

Comment (aabadie) [reply tests/riotboot/tests/01-run.py]: Please squash and no need to retrigger "CI: run tests" just for this.

File: tests/riotboot/Makefile [hunk 3]
```diff
@@ -7,6 +7,8 @@ FEATURES_REQUIRED += riotboot
 # Include modules to test the bootloader
 USEMODULE += riotboot_slot
 USEMODULE += shell
+# Add shell_commands to use shell version off `test_utils_interactive_sync`
```
Comment (aabadie) [inline tests/riotboot/Makefile]:
```suggestion
# Add shell_commands to use the shell version of `test_utils_interactive_sync`
```

File: tests/riotboot/tests/01-run.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2018 Federico Pellegrin <fede@evolware.org>
```
Comment (aabadie) [inline tests/riotboot/tests/01-run.py]:
```suggestion
# Copyright (C) 2019 Francisco Molina <francois-xavier.molina@inria.fr>
```

File: tests/riotboot/tests/01-run.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        "make",
+        "RIOTBOOT_SKIP_COMPILE=1",
+        "riotboot/flash-slot{}".format(slotnum),
+        "APP_VER={}".format(version),
+    ]
+    assert subprocess.call(cmd) == 0
+
+
+def assert_check_slot(child, slotnum, version):
+    # Check if its running on the expected slot
```
Comment (aabadie) [inline tests/riotboot/tests/01-run.py]:
```suggestion
    # Check if it's running on the expected slot
```

**Pull request merged**